Source/FullScreenPokemon.ts

```diff
@@ -2511,6 +2524,8 @@ module FullScreenPokemon {
                     this.StateHolder.addChange(other.id, "sight", undefined);
                 }
             }
+
+            this.autoSave();
```

Comment:
How I understand it, if the `other` is a trainer and hasn't already been battled, the game will start their battle and end up visiting this function again when that's done. Therefore this function will eventually be run with the case that `other.trainer && other.alreadyBattled` is false. 

Is there something missing from that logic?